mlflow/server/js/src/components/artifact-view-components/ShowArtifactImageView.js

```diff
@@ -5,42 +5,67 @@ import { getSrc } from './ShowArtifactPage';
 import Plot from 'react-plotly.js';
 
 class ShowArtifactImageView extends Component {
-  static propTypes = {
-    runUuid: PropTypes.string.isRequired,
-    path: PropTypes.string.isRequired,
-  };
-
   constructor(props) {
     super(props);
     this.state = {
+      loading: true,
       width: 0,
       height: 0,
+      dataURL: '',
+      prevPath: props.path,
     };
   }
 
-  getSrc = () => {
-    const { path, runUuid } = this.props;
-    return getSrc(path, runUuid);
+  static propTypes = {
+    runUuid: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
   };
 
-  resize = () => {
-    const img = new Image();
-    img.src = this.getSrc();
-    img.onload = () => this.setState({ width: img.width, height: img.height });
-  };
+  static getDerivedStateFromProps(props, state) {
+    if (props.path !== state.prevPath) {
+      return {
+        loading: true,
+        prevPath: props.path,
+      };
+    }
+    return null;
+  }
 
   componentDidMount = () => {
-    this.resize();
+    this.fetchImage();
   };
 
   componentDidUpdate = prevProps => {
     if (prevProps.path !== this.props.path) {
-      this.resize();
+      this.fetchImage();
     }
   };
 
+  getSrc = () => {
+    const { path, runUuid } = this.props;
+    return getSrc(path, runUuid);
+  };
+
+  fetchImage = () => {
+    const img = new Image();
+    img.setAttribute('crossOrigin', 'anonymous');
```

Comment:
Could we actually `setState` to `loading` here instead of in `getDerivedStateFromProps`? It's best to avoid the complexity introduced by `getDerivedStateFromProps` when possible, as per https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html

Reply:
I'll fix it. Thanks!

Reply:
@smurching Fixed it